paddle/fluid/framework/ir/mkldnn/reshape_transpose_matmul_mkldnn_fuse_pass.cc

```diff
@@ -153,26 +114,85 @@ void ReshapeTransposeMatmulMkldnnFusePass::Fuse(
 
   gpd(graph, handler);
   AddStatis(found_reshape_transpose_matmul_count);
-  if (!Has("disable_logs") || !Get<bool>("disable_logs")) {
+  if ((!Has("disable_logs") || !Get<bool>("disable_logs")) &&
+      found_reshape_transpose_matmul_count > 0) {
     std::stringstream msg_ss;
-    msg_ss << "---    Fused " << found_reshape_transpose_matmul_count
-           << " ReshapeTransposeMatmul patterns for " << op_name_ << " Op";
+    msg_ss << "---    fused " << found_reshape_transpose_matmul_count
+           << " reshape + transpose + " << matmul_type;
     if (with_reshape_xshape) msg_ss << " with reshape's xshape";
     if (with_transpose_xshape) msg_ss << " with transpose's xshape";
     string::PrettyLogDetail(msg_ss.str().c_str());
   }
 }
 
-void ReshapeTransposeMatmulMkldnnFusePass::ApplyImpl(ir::Graph *graph) const {
-  PADDLE_ENFORCE_NOT_NULL(graph,
-                          platform::errors::InvalidArgument(
-                              "Pointer to graph argument should not be NULL."));
-  FusePassBase::Init(name_scope_, graph);
+ReshapeTransposeMatmulMkldnnFusePass::ReshapeTransposeMatmulMkldnnFusePass() {
+  AddOpCompat(OpCompat("reshape2"))
+      .AddInput("X")
+      .IsTensor()
+      .End()
+      // The reshape2 op for this pass should not have "Shape" and "ShapeTensor"
```

Comment:
The same statement is true for Matmul+Transpose+Reshape fuse pass, but from what I can see it was mistakenly skipped before, could you please fix it in the other pass?

Reply:
Thank you! I wasn't aware of this compat issue